packages/cli/src/util/provider.ts

```diff
@@ -70,27 +70,53 @@ export async function resolveProviderAndSigners({
     throw err;
   }
 
-  const ethersProvider = new CannonWrapperGenericProvider({}, new ethers.providers.Web3Provider(rawProvider as any), false);
+  const localProviders = {
+    frame: 'http://localhost:1248',
+    direct: 'http://localhost:8545',
+  };
+
+  let ethersProvider;
+  const providerList = [];
+
+  // force provider to use JSON-RPC instead of Web3Provider for local instances
+  if (checkProviders.includes('frame') || checkProviders[0].startsWith('http://')) {
```

Comment:
This is actually somewhat wrong still, I dont think we want to setup a JSON-RPC provider for frame EVERYTIME its used, before I was defaulting to that . I think we only want that when connecting to an http endpoint directly so I believe it should just be the this:

```
 // force provider to use JSON-RPC instead of Web3Provider for local instances
  if (checkProviders[0].startsWith('http://')) {
    ethersProvider = new ethers.providers.JsonRpcProvider;
```